server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1457,8 +1459,8 @@ private void writeAtomic(final String blobName, final BytesReference bytesRef, b
 
     @Override
     public void snapshotShard(Store store, MapperService mapperService, SnapshotId snapshotId, IndexId indexId,
-                              IndexCommit snapshotIndexCommit, IndexShardSnapshotStatus snapshotStatus, Version repositoryMetaVersion,
-                              Map<String, Object> userMetadata, ActionListener<String> listener) {
+                              IndexCommit snapshotIndexCommit, boolean isSafeCommit, IndexShardSnapshotStatus snapshotStatus,
```

Comment:
Can we pass the global checkpoint here instead of `isSafeCommit`? Also, can we always store the global checkpoint and the history UUID even when the commit is not safe? If so, I think we can move all the logic to a single place in "findMatchingShardSnapshot"?

Reply:
Yea that's much nicer :) Adjusted the PR accordingly.